src/components/Checkbox/Checkbox.module.scss

```diff
@@ -0,0 +1,59 @@
+@use '../../styles/colors.module';
+
+.container {
+  position: relative;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+}
+
+.hidden {
+  display: none;
+}
+
+.checkbox {
+  $transition: 0.2s ease-in-out;
+
+  position: relative;
+  display: inline-block;
+  width: 1.5rem;
+  height: 1.5rem;
+  margin-right: 1.5rem;
+
+  border: 1px solid var(--color);
+  border-radius: 2px;
+
+  transition: background $transition, border-color $transition;
```

Comment:
Use the mixin, it has additional logic for checking reduced motion and disabling on load to prevent flickering on load